app/controllers/application_controller/compare.rb

```diff
@@ -755,7 +755,7 @@ def comparemiq
     if !session[:checked_items].nil? && @lastaction == "set_checked_items"
       vms = session[:checked_items]
     else
-      vms = find_checked_items
+      vms = find_checked_items_with_rbac(VmOrTemplate)
```

Comment:
Contrary the comment on line 751 and according to lines 762-770, I'd say that this is used not only for VMs but also for other "comparable" entities.

Please, confirm/fix/redo.